public/javascripts/searchBox.js

```diff
@@ -0,0 +1,37 @@
+//this function enable 'enter' on the keyboard to click the submit button
+var enter = function(elem, checkboxClass) {
+    $(elem).keyup(function(e) {
+        if (e.keyCode == 13) {
+            $("#search-button").click()
+        }
+    })
+}
+
+
+var tag = m.prop('')
```

Comment:
consider putting all of these under a function searchBox() {} scope. That way you reduce the risk of variable name conflicts. What if another script defines a var tag?

instead you can do function `searchBoxComponent() {
 var searchBox = {...}
 return {
  searchBox: searchBox
 }
}`